oak-segment-azure/src/main/java/org/apache/jackrabbit/oak/segment/azure/v8/AzurePersistenceV8.java

```diff
@@ -154,4 +155,9 @@ public CloudBlobDirectory getSegmentstoreDirectory() {
     public void setWriteAccessController(WriteAccessController writeAccessController) {
         this.writeAccessController = writeAccessController;
     }
+
+    @TestOnly
+    void disableWriting() {
+        writeAccessController.disableWriting();
+    }
```

Comment:
I would rather not introduce this method. `AzureArchiveManagerV8Test` creates a `WriteAccessController` and passes it to `AzurePersistenceV8`. I think if the test class had a `WriteAccessController` field, `testListArchiveWithDeleteMarkerPresentAndNoWriteAccess()` could disable writes on the `WriteAccessController` instead of `AzurePersistenceV8`.

Reply:
Indeed.